src/components/create-vm-dialog/createVmDialog.jsx

```diff
@@ -195,13 +216,19 @@ function validateParams(vmParams) {
     return validationFailed;
 }
 
-const NameRow = ({ vmName, onValueChanged, validationFailed }) => {
+const NameRow = ({ vmName, suggestedVmName, os, onValueChanged, validationFailed }) => {
     const validationStateName = validationFailed.vmName ? 'error' : 'default';
 
+    let helperText;
+    // Don't show when user inputs any vm name or when we are not able to generate any suggestedVmName from operating system
+    if (isEmpty(vmName.trim()) && (!os || (os && !isEmpty(suggestedVmName.trim()))))
+        helperText = isEmpty(suggestedVmName.trim()) ? _("If blank, a name will be suggested") : _(`If blank, a name will be set to '${suggestedVmName}'`);
```

Comment:
"a name will be suggested" is too surprising. This would only appear until the user selects an OS, right? and that is mandatory? We need to avoid a situation where you click "create" with "a name will be suggested".

Still, I feel this is too magic/jumpy, as a helper text. Easier/less surprising behaviour proposal: set the name input to an autogenerated name once the user chooses an OS and the name is empty still, and then just keep the current "is empty" validation.

Also, grammar: If you keep this sentence, it's "If blank, name will be set.." (drop the "a"). But still, just do it, and avoid the indirection.

As this whole auto-naming business is quite hairy, I also recommend splitting this out into its own PR. I feel it's getting unnecessarily difficult to handle all these independent changes in a single PR.

Reply:
Let's move this whole functionality about generating VM names to a separate PR.